File: Selenium/Activities/Activity14.java
package Example;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
//import java.util.List;

public class Activity14 {
	
	public static void main(String[] args) {
		  
		   WebDriver driver = new FirefoxDriver();
				    
		   driver.get("https://training-support.net/webelements/tables");
				   
		   System.out.println("Title is: "+driver.getTitle());
		
		   List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
		   System.out.println("Number of rows: " +rows.size());
		   
		   List<WebElement> cols = driver.findElements(By.xpath("//table/thead/tr/th"));
		   System.out.println("Number of columns: " +cols.size());
		   
		   WebElement row5 = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
	       System.out.println("Book Name in the 5th row is: " + row5.getText());
	       
	       driver.findElement(By.xpath("//th[text()='Price']")).click();
	       
	       System.out.println("Clicked on header of the Price column");
	       
	       row5 = driver.findElement(By.xpath("//table/tbody/tr[5]/td[2]"));
	       System.out.println("Book Name in the 5th row is: " + row5.getText());
	       
	       driver.quit();
	}
}

File: Selenium/Activities/activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By

with webdriver.Firefox() as driver:
   
   driver.get("https://training-support.net/webelements/tables")
   
   print("Page title is: ", driver.title)
   rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
   print("Number of rows: ",len(rows))

   cols = driver.find_elements(By.XPATH, "//table/thead/tr/th")
   print("Number of cols: ",len(cols))
    
   row5 = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[2]")
   print("Book Name in the 5th row is: ", row5.text)

   driver.find_element(By.XPATH, "//th[text()='Price']").click()

   print("Clicked on header of the Price column")

   row5 = driver.find_element(By.XPATH, "//table/tbody/tr[5]/td[2]")
   print("Book Name in the 5th row is: ", row5.text)


driver.quit()
